specs/templates/table-column-hld.md

```diff
@@ -0,0 +1,134 @@
+# Table Column [Template]
+
+## Overview
```

Comment:
Probably worth linking to https://github.com/ni/nimble/blob/main/packages/nimble-components/src/table/specs/table-columns-hld.md#column-type-philosophy from one or more places in this template. It has useful info on when to create new columns and how to name them.